### PR TITLE
[tgui] Fix usage and update to 0.9.1

### LINKED ISSUE #17056: [TGUI] Install fails without error on macOS 11.1 (20C69)
**Is your feature request related to a problem? Please describe.**

### `CMakeLists.txt`
```cmake
cmake_minimum_required(VERSION 3.0)
cmake_policy(SET CMP0048 NEW)
project(pp VERSION 0.0.0 LANGUAGES CXX)

set(src src/main.cpp)
find_package(TGUI CONFIG REQUIRED)
add_executable(${PROJECT_NAME} ${src})
target_link_libraries(${PROJECT_NAME} PRIVATE tgui)
````

### `src/main.cpp`
First example at the top of https://tgui.eu/tutorials/0.9/sdl-backend/

### `shell`
```sh
$ [path]/vcpkg install tgui
$ mkdir build && cd $_
$ cmake -DCMAKE_TOOLCHAIN_FILE=[path]/vcpkg/scripts/buildsystems/vcpkg.cmake ..
-- The CXX compiler identification is AppleClang 12.0.0.12000032
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at vcpkg/installed/x64-osx/share/tgui/TGUIConfig.cmake:86 (message):
  Requested TGUI configuration (Shared) was not found
Call Stack (most recent call first):
  vcpkg/scripts/buildsystems/vcpkg.cmake:861 (_find_package)
  CMakeLists.txt:6 (find_package)


CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:861 (_find_package):
  Found package configuration file:

    vcpkg/installed/x64-osx/share/tgui/TGUIConfig.cmake

  but it set TGUI_FOUND to FALSE so package "TGUI" is considered to be NOT
  FOUND.
Call Stack (most recent call first):
  CMakeLists.txt:6 (find_package)


-- Configuring incomplete, errors occurred!
See also "/tmp/a/build/CMakeFiles/CMakeOutput.log".
See also "/tmp/a/build/CMakeFiles/CMakeError.log".
```

**Proposed solution**
Would be nice if the example would build

**Describe alternatives you've considered**
I'm trying other cross-platform GUI frameworks that vcpkg 'supports', e.g., see the trackbacks at https://github.com/microsoft/vcpkg/issues/17052

### PR DESCRIPTION
**Pull request by JackBoosY**
1. Fix usage issue:
```
CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:861 (_find_package):
  Found package configuration file:

    vcpkg/installed/x64-osx/share/tgui/TGUIConfig.cmake

  but it set TGUI_FOUND to FALSE so package "TGUI" is considered to be NOT
  FOUND.
```
2. Fix tool `gui-builder`, use the offical commit https://github.com/texus/TGUI/commit/7b1a592223c35feb7e96e23e305134b085b3d7f4.

3. Add feature `tool` and convert dependencies to feature `sdl2` and `sfml`.

4. Add dependencies `opengl` and `glad[egl]` according to the triplet:
_SOURCE/0aa7c449e1-cde739737e/CMakeLists.txt_ ln166:
```cmake
# Enable OpenGL ES on mobile platforms or when explicitly requested (option is ignored by backends that don't use OpenGL)
if(TGUI_OS_IOS OR TGUI_OS_ANDROID)
    set(TGUI_USE_GLES 1)
elseif (NOT DEFINED TGUI_USE_GLES)
    set(TGUI_USE_GLES 0)
endif()
```
_SOURCE/src/Backends/SDL/CMakeLists.txt_ ln24:
```cmake
# Also link to OpenGL or OpenGL ES
if(NOT TGUI_USE_GLES)
    find_package(OpenGL REQUIRED)
    target_link_libraries(tgui PRIVATE OpenGL::GL)
    target_link_libraries(tgui-sdl-interface INTERFACE OpenGL::GL)
else() # OpenGL ES was requested
    # On linux, GL ES 3 code is included in the GLESv2 library
    target_link_libraries(tgui PRIVATE -lEGL -lGLESv2)
    target_link_libraries(tgui-sdl-interface INTERFACE -lEGL -lGLESv2)
endif()
```

Fixes #17056.

Already tested triplets `x86-windows`/`x64-windows-static` with all features.

### CODE REVIEW

**Comment by JackBoosY** [comment]
@SamuelMarks  Can you please test this PR?

Thanks.

**Comment by SamuelMarks** [comment]
Thanks that worked. Just have an `SDL.h` error now, so trying to fix that. Not sure if there should be a help text telling the user they need one of SFML, SDL/OpenGL

**Comment by JackBoosY** [comment]
```
#include <TGUI/Core.hpp>
#include <TGUI/Backends/SDL.hpp>
#include <TGUI/AllWidgets.hpp>
```
These three headers are not belong to tgui.

**Comment by SamuelMarks** [comment]
@JackBoosY Hmm… so my goal is to use TGUI. I look at the examples, bootstrap VCPKG, and expect it to work. Is my workflow misguided?

**Comment by JackBoosY** [comment]
@SamuelMarks  We need to update tgui to latest.

**Comment by JackBoosY** [comment]
@SamuelMarks  Please pull the latest commit and test again.

**Comment by SamuelMarks** [comment]
@JackBoosY Thanks just tried it, but still failing:
```sh
$ /tmp/vcpkg/vcpkg install tgui sdl2 sdl2-ttf
/tmp/tgui $ rm -rf build && mkdir build && cd $_
/tmp/tgui/build $ cmake -DCMAKE_TOOLCHAIN_FILE=/tmp/vcpkg/scripts/buildsystems/vcpkg.cmake ..
-- The CXX compiler identification is AppleClang 12.0.0.12000032
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Freetype: /tmp/vcpkg/installed/x64-osx/lib/libfreetype.a (found version "2.10.4") 
-- Found ZLIB: /tmp/vcpkg/installed/x64-osx/debug/lib/libz.a (found version "1.2.11") 
-- Found BZip2: /tmp/vcpkg/installed/x64-osx/lib/libbz2.a (found version "1.0.8") 
-- Looking for BZ2_bzCompressInit
-- Looking for BZ2_bzCompressInit - found
-- Found PNG: /tmp/vcpkg/installed/x64-osx/debug/lib/libpng.a (found version "1.6.37") 
-- Found SFML 2.5.1 in /tmp/vcpkg/installed/x64-osx/share/sfml
-- Found TGUI 0.9.1 in /tmp/vcpkg/installed/x64-osx/share/tgui
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/tgui/build
/tmp/tgui/build $ cmake --build .
[ 50%] Building CXX object CMakeFiles/pp.dir/src/main.cpp.o
[100%] Linking CXX executable pp
Undefined symbols for architecture x86_64:
  "tgui::GuiSDL::mainLoop()", referenced from:
      run_application(SDL_Window*) in main.cpp.o
  "tgui::GuiSDL::GuiSDL(SDL_Window*)", referenced from:
      run_application(SDL_Window*) in main.cpp.o
  "vtable for tgui::GuiSDL", referenced from:
      tgui::GuiSDL::~GuiSDL() in main.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [pp] Error 1
make[1]: *** [CMakeFiles/pp.dir/all] Error 2
make: *** [all] Error 
```

**Comment by JackBoosY** [comment]
@SamuelMarks  Okay, we need to add dependency sdl2.

**Comment by JackBoosY** [comment]
Since I met some problems when building `gui-builder` (https://github.com/texus/TGUI/issues/157), disable it temporary.

**Comment by SamuelMarks** [comment]
Sounds like a plan.

RE: dependencies. I think SDL2 is optional, as you have a choice betwixt that and SGML. So it might be more of a `vcpkg install tgui[sdl2]` type thing à la imgui.

**Comment by JackBoosY** [comment]
@SamuelMarks Yes, `sdl2` and `sfml` are features of tgui, and I set them to default features.
Since `sdl2` has conflict with `gui-builder`, according to our policy, I disabled it temporary until the upstream solve it.

**Review by ytbigvinny27 — APPROVED** [review]

File: versions/t-/tgui.json [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "6d443ef6067d1364bbb832a041ed6cf4636b3dd8",
```
Comment (JackBoosY) [inline versions/t-/tgui.json:4]:
```suggestion
      "git-tree": "d1bec343af21cd9538526ae2d3abc74a2585be36",
```

File: ports/tgui/portfile.cmake [hunk 2]
```diff
@@ -53,8 +54,12 @@ if(BUILD_GUI_BUILDER)
     endif()
 endif()
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include ${CURRENT_PACKAGES_DIR}/debug/share)
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/include/TGUI/nanosvg")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include" "${CURRENT_PACKAGES_DIR}/debug/share")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/include/TGUI/nanosvg" "${CURRENT_PACKAGES_DIR}/include/TGUI/extlibs/glad"
+    "${CURRENT_PACKAGES_DIR}/include/TGUI/extlibs/nanosvg" "${CURRENT_PACKAGES_DIR}/include/TGUI/extlibs/stb"
+)
```
Comment (PhoebeHui) [inline ports/tgui/portfile.cmake:60]: I'm concerned if tgui builds these external libraries or depends on them? if remove them, we didn't add any dependency to tgui, users may encounter the usage issues.

Comment (texus) [reply ports/tgui/portfile.cmake:60]: These are all header-only libraries (some of them modified) that are mostly needed when building the library. Several of the extlibs folders can be removed without breaking user code. I'm curious for the reason why they were being removed though (and risking potentially removing too much).

There are currently 4 extlibs:
- Aurora is exposed in public headers and is the only extlib that can't be removed
- glad is included in a public header file, but this header file is undocumented so it is unlikely to be included in user code
- nanosvg is only included internally
- stb is only included internally

**Comment by SamuelMarks** [comment]
FYI: At latest commit here TGUI works well (tested with SDL backend) 🎉

**Comment by SamuelMarks** [comment]
FYI: At latest commit here TGUI works well (tested with SDL backend) 🎉

BTW: I did come across `ld: warning: direct access… `libtgui-s-d.a(BackendSDL.cpp.o)' to global weak symbol 'typeinfo for tgui::Exception'` but I think it's a TGUI issue rather than a `vcpkg` one, so I posted https://github.com/texus/TGUI/issues/158

**Comment by SamuelMarks** [comment]
FYI: At latest commit here TGUI works well (tested with SDL backend) 🎉

BTW: I did come across
> ld: warning: direct access… `libtgui-s-d.a(BackendSDL.cpp.o)' to global weak symbol 'typeinfo for tgui::Exception'`

…but I think it's a TGUI issue rather than a `vcpkg` one, so I posted https://github.com/texus/TGUI/issues/158

File: ports/tgui/portfile.cmake (hunk 2 above)
Comment (JackBoosY) [reply ports/tgui/portfile.cmake:60]: @PhoebeHui I confirmed that all the removed headers are same with the dependencies.

**Comment by SamuelMarks** [comment]
@JackBoosY Actually can you merge in the latest version? - 7 hours ago the BDFL posted https://github.com/texus/TGUI/commit/6515153c2466e6677ba933679b3dca6c283daf87 which solved that bug I commented about here

File: ports/tgui/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      "description": "build the SDL backend",
+      "dependencies": [
+        "sdl2",
+        "sdl2-ttf"
+      ]
+    },
+    "sfml": {
+      "description": "build the SFML backend",
+      "dependencies": [
+        "sfml"
```
Comment (JackBoosY) [inline ports/tgui/vcpkg.json:21]:
```suggestion
        "sfml".
        "stb",
        {
            "name": "glad[egl]",
            "platform": "!android & !ios"
        },
        {
            “name”:: "opengl".
            "platform": "android | ios"
        }
```

File: ports/tgui/portfile.cmake (hunk 2 above)
Comment (JackBoosY) [reply ports/tgui/portfile.cmake]: Now, the extra dependencies header folders are all empty, just remove them.

**Comment by JackBoosY** [comment]
@SamuelMarks  Done.

File: versions/baseline.json [hunk 4]
```diff
@@ -5869,7 +5869,7 @@
       "port-version": 0
     },
     "tgui": {
-      "baseline": "0.8.8",
+      "baseline": "0.9.1",
```
Comment (JackBoosY) [inline versions/baseline.json:5872]:
```suggestion
      "baseline": "2021-04-19",
```

**Comment by JackBoosY** [comment]
Already test all features on `x86-windows` and `x64-windows-static`.

**Comment by JackBoosY** [comment]
@SamuelMarks  Can you please test my changes?

**Review by PhoebeHui — APPROVED** [review]
LGTM.

**Pull request merged**